reference/src/layout/arrays.md

```diff
@@ -0,0 +1,33 @@
+# Layout of Rust array types
+
+Array types, `[T; N]`, store `N` values of type `T` contiguously with a constant
+_stride_ (the distance between each two consecutive values).
+
+The offset of the first array element is `0`.
+
+The stride of the array is computed as the size of the element type rounded up
+to the next multiple of the alignment of the element type. That is, the _stride_
+of an array can be larger than the element size iff the alignment requirements
+of the element are larger than its size.
+
+Having a stride larger than the element size allows:
+
+```rust,ignore
+struct A(u16, u8); // size_of == 3, align_of == 4
+type B = [A; 4];  // => stride == 4 > 3
+```
+
+The size and alignment of `Vector` element types match, such that `Vector` types
+and arrays are layout compatible.
+
+`repr(C)` arrays have the same layout as C arrays and are passed by pointer in C
+FFI according to the C ABI.
```

Comment:
This is just a consequence of the prior statements, not a new rule, correct? That should be made more explicit.

Reply:
Let's continue this discussion below (i've cc'ed you).